nitrite.go

```diff
@@ -342,20 +362,28 @@ func Verify(data []byte, options VerifyOptions) (*Result, error) {
 	if nil == roots {
 		roots = defaultRoot
 	}
+	if cert.IsCA {
+		roots.AddCert(cert)
+	}
 
 	currentTime := options.CurrentTime
 	if currentTime.IsZero() {
 		currentTime = time.Now()
 	}
 
-	_, err = cert.Verify(x509.VerifyOptions{
-		Intermediates: intermediates,
-		Roots:         roots,
-		CurrentTime:   currentTime,
-		KeyUsages: []x509.ExtKeyUsage{
-			x509.ExtKeyUsageAny,
+	_, err = cert.Verify(
+		x509.VerifyOptions{
+			Intermediates: intermediates,
+			Roots:         roots,
+			CurrentTime:   currentTime,
+			KeyUsages: []x509.ExtKeyUsage{
+				x509.ExtKeyUsageAny,
+			},
 		},
-	})
+	)
+	if err != nil {
```

Comment:
Probably because they are not running a linter like we do in zpr! Indeed, I just ran the linter and it still found two small (and most likely inconsequential) things:
```golang
nitrite.go:185:6: func `reverse` is unused (unused)
func reverse(enc []byte) []byte {
     ^
nitrite.go:294:6: SA4003: no value of type `uint` is less than `0` (staticcheck)
		if key < 0 || key > 31 {
		
```

Might I suggest you add a basic Makefile with a command for running the linter (example from zpr below). You might also consider adding that to the github pipeline for this repo

```
lint:
	@golangci-lint run --config ./configs/golangci.yaml
```

Reply:
Honestly, I think we should consider adding unit tests for `nitrite.go` as well. The repo is small, but given the importance of verifying nitro attestations it seems bad to not have any testing on it. If you think this is out-of-scope for the PR then I suggest adding a new card for adding some testing. Might be a good card for me to take on to better familiarize myself with the code

Reply:
I added a `lint` target and fixed the errors.

I agree that we should have tests in `nitrite` and to do so, we need to make quite a bit of refactoring. Doing so was originally on the list, but as we became more focused on getting the sequencer up, we removed it from the sequencer skeleton epic. In fact, we should refactor and test not only `nitrite`, but also `viproxy`, `socksproxy`, and `nsm` dependencies. This one is pretty important though, so I made a card that may become an epic: https://blocky.atlassian.net/browse/BKY-3396